pkg/services/apiserver/storage/entity/storage.go

```diff
@@ -442,6 +442,11 @@ func (s *Storage) GetList(ctx context.Context, key string, opts storage.ListOpti
 	if len(requirements.SortBy) > 0 {
 		req.Sort = requirements.SortBy
 	}
+	if len(requirements.ListOriginKey) > 0 {
+		// #TODO: figure out if there is support for multiple keys since
+		// the doc comments mention "limit to a specific key"
+		req.Key = requirements.ListOriginKey
```

Comment:
Afaik these values would be grafana uuids, not k8s keys. We'll need to extend EntityListRequest so that it can accept a new OriginKey list, and extend entityStoreServer.List to translate that into a where condition if it's specified.

Reply:
Not sure if this is what you meant by the update to entityStoreServer.List, but this seemed like the most appropriate place: https://github.com/grafana/grafana/blob/5f62662e10755b0e63912e6ab3da51e4716e586d/pkg/services/store/entity/sqlstash/sql_storage_server.go#L1303-L1307